core/src/main/java/hudson/model/RunMap.java

```diff
@@ -185,7 +185,7 @@ public R put(R r) {
         // Defense against JENKINS-23152 and its ilk.
         File rootDir = r.getRootDir();
         if (rootDir.isDirectory()) {
-            throw new IllegalStateException(rootDir + " already existed; will not overwrite with " + r);
+            throw new IllegalStateException("JENKINS-23152: " + rootDir + " already existed; will not overwrite with " + r);
```

Comment:
Just to make sure people encountering these exceptions have a trail to the filed issues.